src/JavascriptVM/vm.ts

```diff
@@ -233,7 +233,12 @@ export class BlocklyInterpreter {
       );
     });
 
+    // this may cause an infinite loop, we should bail out if this is the case
```

Comment:
I am afraid this could be rather common, e.g.:

<img width="404" alt="unresponsive" src="https://user-images.githubusercontent.com/15800423/128610265-5d853500-9537-4566-a6ba-8e2cb0dcf177.png">

Would it be possible to step() the code until the JS-Interpreter's `step()` method returns false? We could use a counter to "bail out" if we haven't managed to parse the code to completion within a reasonable amount of steps?

Reply:
i've managed to fix this, now only code in Start will be generated. I still need to handle the case where there is no start. Ive put some tokens in the generated javascript to detect later.